edge-modules/azure-monitor/src/IotHubUpload/IotHubUpload.cs

```diff
@@ -42,6 +42,11 @@ public async Task<bool> PublishAsync(IEnumerable<Metric> metrics, CancellationTo
                 }
 
                 byte[] metricsData = Encoding.UTF8.GetBytes(outputString);
+                if (Settings.Current.CompressForUpload)
+                {
+                    metricsData = Compression.CompressToGzip(metricsData);
```

Comment:
For context, the azure monitor upload does not use gzip compression. Why did they make a different decision than gzip?

Reply:
Probably they align with what ever compressed format that the cloud-side service accepts. We are not changing that flow right and this is only for UploadTarget = 'IoTMessage' flow?

Reply:
Yes we are only changing the iothub upload flow